PocketKit/Tests/PocketKitTests/Refresh/HomeRefreshCoordinatorTests.swift

```diff
@@ -0,0 +1,134 @@
+import XCTest
+import Sync
+import Combine
+
+@testable import PocketKit
+
+class HomeRefreshCoordinatorTests: XCTestCase {
+    private var notificationCenter: NotificationCenter!
+    private var userDefaults: UserDefaults!
+    private var source: MockSource!
+    private var subscriptions: Set<AnyCancellable>!
+
+    override func setUpWithError() throws {
+        notificationCenter = NotificationCenter()
+        userDefaults = UserDefaults()
+        source = MockSource()
+        subscriptions = []
+    }
+
+    override func tearDownWithError() throws {
+        userDefaults.removeObject(forKey: HomeRefreshCoordinator.dateLastRefreshKey)
+    }
+
+    func subject(
+        notificationCenter: NotificationCenter? = nil,
+        userDefaults: UserDefaults? = nil,
+        source: Source? = nil,
+        refreshRuleTime: Int = 1
+    ) -> HomeRefreshCoordinator {
+        HomeRefreshCoordinator(
+            notificationCenter: notificationCenter ?? self.notificationCenter,
+            userDefaults: userDefaults ?? self.userDefaults,
+            source: source ?? self.source,
+            refreshRuleTime: refreshRuleTime
+        )
+    }
+
+    func test_firstRefresh_setsUserDefaults() {
+        XCTAssertNil(userDefaults.object(forKey: HomeRefreshCoordinator.dateLastRefreshKey))
+        source.stubFetchSlateLineup { _ in }
+
+        let expectRefresh = expectation(description: "Refresh home")
+
+        let coordinator = subject()
+
+        coordinator.refresh {
+            expectRefresh.fulfill()
+        }
+        wait(for: [expectRefresh], timeout: 1)
+        XCTAssertNotNil(userDefaults.object(forKey: HomeRefreshCoordinator.dateLastRefreshKey))
+    }
+
+    func test_coordinator_whenDataIsNotStale_doesNotRefreshHome() {
+        let date = Date()
+        userDefaults.setValue(date, forKey: HomeRefreshCoordinator.dateLastRefreshKey)
+        XCTAssertNotNil(userDefaults.object(forKey: HomeRefreshCoordinator.dateLastRefreshKey))
+        source.stubFetchSlateLineup { _ in }
+
+        let coordinator = subject()
+        coordinator.refresh {
+            XCTFail("Should not have refreshed")
+        }
+        XCTAssertEqual(userDefaults.object(forKey: HomeRefreshCoordinator.dateLastRefreshKey) as? Date, date)
+    }
+
+    func test_coordinator_whenDataIsStale_refreshesHome() {
+        let date = Calendar.current.date(byAdding: .hour, value: -2, to: Date())
```

Comment:
Should this be -2, or -12?

Reply:
Ah, I set the `refreshRuleTime` in the tests to be 1 instead of 12, but I can see how that can be confusing, so I'll update it to 12.